provider/k8s.go

```diff
@@ -372,10 +374,12 @@ type BlobStore interface {
 	Writer(key string) (*blob.Writer, error)
 	Read(key string) ([]byte, error)
 	Serve(key string) (Iterator, error)
+	ServeDirectory(dir string) (Iterator, error)
 	Exists(key string) (bool, error)
 	Delete(key string) error
 	DeleteAll(dir string) error
 	NewestBlob(prefix string) string
+	SparkOutputPartList(prefix string) []string
```

Comment:
Dont do this. If spark needs additional functions for itself then create another interface that implements this interface. Functions in this interface should only be relevant to the blob store

Reply:
true, this feature creep is why the other interface ended up so bloated.